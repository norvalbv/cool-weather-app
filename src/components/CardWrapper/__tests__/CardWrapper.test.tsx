import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import CardWrapper, { CardWrapperProps } from '..';

const CardWrapperWithAllProps: FC<Required<CardWrapperProps>> = CardWrapper;

describe('<CardWrapper />', () => {
  test('Should render correctly with mandatory props', () => {
    const tree = renderer
      .create(
        <CardWrapper>
          <div>Child</div>
        </CardWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with all props', () => {
    const tree = renderer
      .create(
        <CardWrapperWithAllProps
          backgroundColor="white"
          cardType="default"
          className="bg-red-500"
          role="button"
        >
          <div>Child</div>
        </CardWrapperWithAllProps>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
