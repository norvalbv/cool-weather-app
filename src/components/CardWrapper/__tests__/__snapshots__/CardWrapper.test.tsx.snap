// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWrapper /> Should render correctly with all props 1`] = `
<div
  className="relative z-10 bg-white text-black"
  role="button"
>
  <div
    className="max-w-[2000px] mx-auto bg-red-500"
  >
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`<CardWrapper /> Should render correctly with mandatory props 1`] = `
<div
  className="relative z-10 bg-primary-dark text-primary-light"
>
  <div
    className="max-w-[2000px] mx-auto"
  >
    <div>
      Child
    </div>
  </div>
</div>
`;
